src/pdsfile/holdings_maintenance/_linkshelf_common.py

```
@@ -0,0 +1,712 @@
+def link_update(spec, pdsdir, *, logger=None, limits=None):
+    """Add the links of any file the shelf does not already carry."""
+
+    if limits is None:
+        limits = {}
+
+    link_path = pdsdir.shelf_path_and_lskip('link')[0]
+
+    # Make sure link shelf file exists
+    if not os.path.exists(link_path):
+        logger = logger or pdslogger.PdsLogger.get_logger(spec.logname)
+        logger.warning('Link shelf file does not exist; initializing', link_path)
```

Ruff (0.16.1)

[error] Too many arguments for `logging` format string

(PLE1205)